pandas/core/arrays/array_.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+
+from pandas._libs import lib, tslibs
+
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+from pandas import compat
+
+
+def array(data, dtype=None, copy=False):
+    """
+    Create an array.
+
+    .. versionadded:: 0.24.0
+
+    Parameters
+    ----------
+    data : Sequence[object]
+        A sequence of objects. The scalars inside `data` should
+        be instances of the scalar type for `dtype`.
+
+        When `data` is an Index or Series, the underlying array
+        will be extracted from `data`.
+
+    dtype : Union[str, np.dtype, ExtensionDtype], optional
+        The dtype to use for the array. This may be a NumPy
+        dtype or an extension type registered with pandas using
+        :meth:`pandas.api.extensions.register_extension_dtype`.
+
+        If not specified, there are two possibilities:
+
+        1. When `data` is a :class:`Series`, :class:`Index`, or
+           :class:`ExtensionArray`, the `dtype` will be taken
+           from the data.
+        2. Otherwise, pandas will attempt to infer the `dtype`
+           from the data.
+
+        Note that when `data` is a NumPy array, ``data.dtype`` is
+        ignored.
+
+    copy : bool, default False
+        Whether to copy the data.
+
+    Returns
+    -------
+    array : Union[numpy.ndarray, ExtensionArray]
+
+    Notes
+    -----
+    Omitting the `dtype` argument means pandas will attempt to infer the
+    best array type from the values in the data. As new array types are
+    added by pandas and 3rd party libraries, the best array type may
+    change. We recommend specifying `dtype` to ensure that the correct
```

Comment:
```suggestion
    change. We recommend specifying `dtype` to ensure that a stable and the correct
```

My suggestion is maybe not fully a proper sentence construction, but wanted to have some notion in it about if you want a stable result across versions. 
(but thanks for adding the note!)